riscv/processor.h

```diff
@@ -300,12 +325,10 @@ class processor_t : public abstract_device_t
     // variable xlen, this method should be removed.
     return xlen;
   }
-  unsigned get_max_xlen() { return max_xlen; }
-  std::string get_isa_string() { return isa_string; }
   unsigned get_flen() {
     return extension_enabled('Q') ? 128 :
-           extension_enabled('D') ? 64 :
-           extension_enabled('F') ? 32 : 0;
+           extension_enabled('D') || extension_enabled(EXT_ZDINX) ? 64 :
+           extension_enabled('F') || extension_enabled(EXT_ZFINX) ? 32 : 0;
```

Comment:
This doesn't make sense.  FLEN refers to the width of an F-register.  For Zfinx and Zdinx, there are no F-registers, and therefore it doesn't make sense to talk about their length.

Reply:
The FLEN is also used in Zfinx. It refers to the width of virual 'F-register'(which store the value in integer register or registers, such as https://github.com/riscv/riscv-zfinx/blob/main/Zfinx_spec.adoc#rv32fd-zfh-ZFinx-register-pair-handling).

Reply:
That's not actually the Zfinx spec; that is an out-of-date proposal.  The actual spec is here https://github.com/riscv/riscv-zfinx/blob/main/zfinx-1.0.0-rc.pdf and makes no mention of the term FLEN.  FLEN still means "the width of an F register".

Reply:
Ok. I'll try to change this.

Reply:
Hi @aswaterman there's no defined discovery mechanism yet, and certainly nothing in the Z[fd]inx spec about detecting which of these two is implemented. I agree that FLEN is the incorrect term as there are no F-registers, but the model does need to know which of these two are implemented, so how should it distinguish between them? I suppose setting a Zfinx_valid / Zdinx_valid flag would do it.

Reply:
The discovery mechanism is irrelevant here, because we're talking about a data structure internal to Spike.  `processor_t::extension_enabled(EXT_ZFINX)` suffices - that's equivalent to your "zfinx_valid" flag.

Reply:
@aswaterman Do I need to revert the unified ISA-string processing for spike-dasm and spike?